File: sample/todo-app-ios/todo-app-ios/details/TodoDetailsViewProxy.swift
//
//  TodoDetailsViewImpl.swift
//  todo-app-ios
//
//  Created by Arkadii Ivanov on 4/4/20.
//  Copyright © 2020 arkivanov. All rights reserved.
//

import Foundation
import TodoLib

class TodoDetailsViewProxy: BaseMviView<TodoDetailsViewModel, TodoDetailsViewEvent>, TodoDetailsView, ObservableObject {
    
    @Published var model: TodoDetailsViewModel?
    
    override func render(model: TodoDetailsViewModel) {
        self.model = model
    }
}

File: sample/todo-app-js/src/main/kotlin/details/TodoDetailsViewProxy.kt
package details

import com.arkivanov.mvikotlin.core.view.BaseMviView
import com.arkivanov.mvikotlin.sample.todo.common.view.TodoDetailsView

class TodoDetailsViewProxy(
    private val updateState: (TodoDetailsView.Model) -> Unit
) : BaseMviView<TodoDetailsView.Model, TodoDetailsView.Event>(), TodoDetailsView {

    override fun render(model: TodoDetailsView.Model) {
        updateState(model)
    }
}
